packages/flutter_test/lib/src/test_text_input_key_handler.dart

```diff
@@ -0,0 +1,158 @@
+    const SingleActivator(LogicalKeyboardKey.keyP, control: true): <String>['moveUp:'],
+    const SingleActivator(LogicalKeyboardKey.keyP, control: true, shift: true): <String>['moveUpAndModifySelection:'],
+    const SingleActivator(LogicalKeyboardKey.keyT, control: true): <String>['transpose:'],
+    const SingleActivator(LogicalKeyboardKey.keyV, control: true): <String>['pageDown:'],
+    const SingleActivator(LogicalKeyboardKey.keyV, control: true, shift: true): <String>['pageDownAndModifySelection:'],
+    const SingleActivator(LogicalKeyboardKey.keyY, control: true): <String>['yank:'],
+    const SingleActivator(LogicalKeyboardKey.quoteSingle, control: true): <String>['insertSingleQuoteIgnoringSubstitution:'],
+    const SingleActivator(LogicalKeyboardKey.quote, control: true): <String>['insertDoubleQuoteIgnoringSubstitution:'],
+    const SingleActivator(LogicalKeyboardKey.home): <String>['scrollToBeginningOfDocument:'],
+    const SingleActivator(LogicalKeyboardKey.end): <String>['scrollToEndOfDocument:'],
+    const SingleActivator(LogicalKeyboardKey.home, shift: true): <String>['moveToBeginningOfDocumentAndModifySelection:'],
+    const SingleActivator(LogicalKeyboardKey.end, shift: true): <String>['moveToEndOfDocumentAndModifySelection:'],
+    const SingleActivator(LogicalKeyboardKey.pageUp): <String>['scrollPageUp:'],
+    const SingleActivator(LogicalKeyboardKey.pageDown): <String>['scrollPageDown:'],
+    const SingleActivator(LogicalKeyboardKey.pageUp, shift: true): <String>['pageUpAndModifySelection:'],
+    const SingleActivator(LogicalKeyboardKey.pageDown, shift: true): <String>['pageDownAndModifySelection:'],
+    const SingleActivator(LogicalKeyboardKey.escape): <String>['cancelOperation:'],
+    const SingleActivator(LogicalKeyboardKey.enter): <String>['insertNewline:'],
+    const SingleActivator(LogicalKeyboardKey.enter, alt: true): <String>['insertNewlineIgnoringFieldEditor:'],
+    const SingleActivator(LogicalKeyboardKey.enter, control: true): <String>['insertLineBreak:'],
+    const SingleActivator(LogicalKeyboardKey.tab): <String>['insertTab:'],
+    const SingleActivator(LogicalKeyboardKey.tab, alt: true): <String>['insertTabIgnoringFieldEditor:'],
+    const SingleActivator(LogicalKeyboardKey.tab, shift: true): <String>['insertBacktab:'],
+  };
+
+  @override
+  Future<void> handleKeyDownEvent(LogicalKeyboardKey key) async {
+    if (key == LogicalKeyboardKey.shift ||
+        key == LogicalKeyboardKey.shiftLeft ||
+        key == LogicalKeyboardKey.shiftRight) {
+      _shift = true;
+    } else if (key == LogicalKeyboardKey.alt ||
+        key == LogicalKeyboardKey.altLeft ||
+        key == LogicalKeyboardKey.altRight) {
+      _alt = true;
+    } else if (key == LogicalKeyboardKey.meta ||
+        key == LogicalKeyboardKey.metaLeft ||
+        key == LogicalKeyboardKey.metaRight) {
+      _meta = true;
+    } else {
```

Comment:
Why no controls?

Reply:
Seems like an oversight. Also this should have caught by tests. I'll look into this.

Reply:
Indeed, control was missing :-/ The transpose test didn't fail, because, because Control+T did result in `transpose:` selector selector. The issue is that without checking for control Just `T` itself would result in `transpose:` as well. Fixed.